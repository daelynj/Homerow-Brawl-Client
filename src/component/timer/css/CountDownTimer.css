.countdown {
  font-size: 20pt;
  margin-left: auto;
  margin-right: auto;
  width: 30%;
  border: 5px solid red;
  text-align: center;
}
