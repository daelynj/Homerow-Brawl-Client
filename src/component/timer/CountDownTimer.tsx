import * as React from "react";
import { useState, useEffect } from "react";
import "./css/CountDownTimer.css";

interface Props {
  setCountUp: (newCountUp: boolean) => void;
}

export const CountDownTimer = (props: Props) => {
  const [seconds, setSeconds] = useState<number>(5);

  useEffect(() => {
    let timerID = setTimeout(tick, 1000);
  });

  const tick = () => {
    if (seconds > 1) {
      setSeconds(seconds - 1);
    } else {
      props.setCountUp(true);
    }
  };

  return (
    <div className="countdown">
      <div>Get ready to type!</div>
      Timer: {seconds}
    </div>
  );
};
