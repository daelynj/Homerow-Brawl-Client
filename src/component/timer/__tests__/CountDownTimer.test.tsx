import * as React from "react";
import { CountDownTimer } from "../CountDownTimer";
import renderer from "react-test-renderer";

describe("CountDownTimer", () => {
  jest.useFakeTimers();
  const setCountUp = jest.fn();

  it("renders the CountDownTimer", () => {
    const tree = renderer
      .create(<CountDownTimer setCountUp={setCountUp} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls tick every 1000 miliseconds", () => {
    const TimerComponent = renderer.create(
      <CountDownTimer setCountUp={setCountUp} />
    );

    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 1000);

    TimerComponent.update(<CountDownTimer setCountUp={setCountUp} />);

    expect(setTimeout).toHaveBeenCalledWith(expect.any(Function), 1000);
  });
});
