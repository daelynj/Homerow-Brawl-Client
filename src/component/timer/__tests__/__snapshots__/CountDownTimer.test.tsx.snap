// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountDownTimer renders the CountDownTimer 1`] = `
<div
  className="countdown"
>
  <div>
    Get ready to type!
  </div>
  Timer: 
  5
</div>
`;
